File: packages/sp_shared/lib/src/models/pattern_summary.dart
/// Summary of a pattern from root meta.json.
///
/// Used in browse lists and for cache invalidation.
final class PatternSummary {
  const PatternSummary({
    required this.id,
    required this.dataVersion,
    required this.displayName,
    required this.feedUrlHint,
    required this.playlistCount,
  });

  factory PatternSummary.fromJson(Map<String, dynamic> json) {
    return PatternSummary(
      id: json['id'] as String,
      dataVersion: (json['dataVersion'] as int?) ?? 1,
      displayName: json['displayName'] as String,
      feedUrlHint: json['feedUrlHint'] as String,
      playlistCount: json['playlistCount'] as int,
    );
  }

  final String id;
  final int dataVersion;
  final String displayName;
  final String feedUrlHint;
  final int playlistCount;

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'dataVersion': dataVersion,
      'displayName': displayName,
      'feedUrlHint': feedUrlHint,
      'playlistCount': playlistCount,
    };
  }
}

File: crates/sp_core/src/models/pattern_summary.rs
use serde::{Deserialize, Serialize};

use super::default_data_version;

/// Summary of a pattern from root meta.json.
#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PatternSummary {
    pub id: String,

    #[serde(default = "default_data_version")]
    pub data_version: i32,

    pub display_name: String,
    pub feed_url_hint: String,
    pub playlist_count: i32,
}
